addOns/exim/src/test/java/org/zaproxy/addon/exim/sites/SiteTreeHandlerUnitTest.java

```diff
@@ -513,4 +546,151 @@ void shoulPrubeDdnNode() throws Exception {
         // And that the node hierarchy really was deleted
         assertThat(siteMap.getRoot().getChildCount(), is(0));
     }
+
+    @Test
+    public void shouldHandleMultilineValues() throws Exception {
+        // Given
+        String multilineValue = "line1\nline2\nline3";
+
+        // Create a POST message with multiline parameters
+        HttpMessage msg = new HttpMessage(new URI("https://www.example.com/multiline", true));
+        msg.getRequestHeader().setMethod(HttpRequestHeader.POST);
+        msg.getRequestHeader()
+                .setHeader(HttpHeader.CONTENT_TYPE, "application/x-www-form-urlencoded");
+        msg.setRequestBody("multiline=" + multilineValue);
+
+        // Configure mocks
+        HistoryReference href = getHref(msg);
+        given(href.getURI()).willReturn(new URI("https://www.example.com/multiline", true));
+        given(href.getMethod()).willReturn(HttpRequestHeader.POST);
+        given(href.getHttpMessage()).willReturn(msg);
+        given(href.getStatusCode()).willReturn(200);
+        given(href.getResponseHeaderLength()).willReturn(20);
+        given(href.getResponseBodyLength()).willReturn(30);
+
+        // Set up test parameters
+        List<NameValuePair> params = new ArrayList<>();
+        params.add(
+                mock(
+                        NameValuePair.class,
+                        invocation -> {
+                            Method method = invocation.getMethod();
+                            if (method.getName().equals("getName")) {
+                                return "multiline";
+                            } else if (method.getName().equals("getValue")) {
+                                return multilineValue;
+                            }
+                            return null;
+                        }));
+        given(session.getParameters(any(HttpMessage.class), eq(HtmlParameter.Type.form)))
+                .willReturn(params);
+
+        // Add to site map
+        SiteNode node = new SiteNode(null, 1, "POST:/multiline");
+        node.setHistoryReference(href);
+        siteMap.getRoot().add(node);
+
+        // When
+        StringWriter sw = new StringWriter();
+        ExporterResult result = new ExporterResult();
+        SitesTreeHandler.exportSitesTree(sw, siteMap, result);
+
+        // Then
+        String output = sw.toString();
+        assertThat(output.contains("multiline="), is(true));
+    }
+
+    @Test
+    public void shouldHandleSpecialCharacters() throws Exception {
```

Comment:
This is not testing the handling of the special characters, the value of the parameters is not persisted.